sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -120,7 +120,7 @@ private[sql] object JDBCRDD extends Logging {
     val dialect = JdbcDialects.get(url)
     val conn: Connection = getConnector(properties.getProperty("driver"), url, properties)()
     try {
-      val rs = conn.prepareStatement(s"SELECT * FROM $table WHERE 1=0").executeQuery()
+      val rs = conn.createStatement.executeQuery(s"SELECT * FROM $table WHERE 1=0")
```

Comment:
Ah hang on a sec, I see you've changed some non-DDL statements in this version of the PR. The change is just about avoiding `PreparedStatement` for DDL right? My earlier comments were about closing resources for a bunch of queries including non-DDL. But if you're only addressing DDL, then your last PR was fine (before it was corrupted).